arrows/qt/widgets/MetadataView.cxx

```diff
@@ -30,135 +30,150 @@ namespace kwiver {
 namespace arrows {
 namespace qt {
 
-QTE_IMPLEMENT_D_FUNC(MetadataView)
+QTE_IMPLEMENT_D_FUNC( MetadataView )
 
 ///////////////////////////////////////////////////////////////////////////////
 
 //BEGIN MetadataViewPrivate
 
-//-----------------------------------------------------------------------------
+// ----------------------------------------------------------------------------
 class MetadataViewPrivate
 {
 public:
-  void addItem(int id, QString const& keyText);
-  void setItemValue(int id, QString const& valueText);
-  void clearItemValue(int id);
+  void addItem( int id, QString const& keyText );
+  void setItemValue( int id, QString const& valueText );
+  void clearItemValue( int id );
 
-  void removeItem(int id);
+  void removeItem( int id );
   void clear();
 
-  QSet<int> itemIds() const { return this->m_keyLabels.keys().toSet(); }
+  QSet< int >
+  itemIds() const { return this->m_keyLabels.keys().toSet(); }
   void updateLabelColors();
 
   QWidget* m_contentWidget;
 
 protected:
-  static void setKeyLabelColor(QLabel* label);
+  static void setKeyLabelColor( QLabel* label );
 
-  QHash<int, QLabel*> m_keyLabels;
-  QHash<int, qtSqueezedLabel*> m_valueLabels;
+  QHash< int, QLabel* > m_keyLabels;
+  QHash< int, qtSqueezedLabel* > m_valueLabels;
 };
 
-//-----------------------------------------------------------------------------
-void MetadataViewPrivate::addItem(int id, QString const& keyText)
+// ----------------------------------------------------------------------------
+void
+MetadataViewPrivate
+::addItem( int id, QString const& keyText )
 {
-  if (auto* const l = this->m_keyLabels.value(id, nullptr))
+  if ( auto* const l = this->m_keyLabels.value( id, nullptr ) )
```

Comment:
Aaaaaack... ***so*** sorry for not noticing this before, but I believe there is not supposed to be a space in `if(`. That's my bad for botching the uncrustify configuration; `sp_before_sparen` should be `remove`, not `force`.

...though OTOH I'd be willing to discuss that aspect of the preferred style. I'd also be quite willing to go ahead and merge this without that change and circle back to it.

Reply:
I suggest we merge what we have, since this is what uncrustify provides right now and is better than it was.  You can improve uncrustify and apply again in a new branch.

Reply:
Right, that's what I said. But please don't delete the correct blank line at L293.

(Just to be clear, though, this isn't an uncrustify bug, it's my fault for *asking* it to do the wrong thing. Fixing it is just a matter of fixing [the config](https://gist.github.com/mwoehlke-kitware/4a2a2e73189928ffb74a1d79e989eb29), which I've done in the latest version thereof.)

Reply:
done.

You should contribute this uncrustify config file to KWIVER with some documentation about how to use it (e.g. in `CONTRIBUTING.rst`).  Otherwise no one will use it.  I understand it's not perfect, but it seems fairly close.

Reply:
Honestly, it... depends. You lucked out in that the only "malformatting" that seems to occur in these files is a known/expected issue due to a missing feature and not a bug. Even in the rest of `arrows/qt`, that's less true. For example, one of the issues I'm having is that it wants to add a newline between the type and method name for inline methods, which I believe we don't want. It's also doing some funky things to certain declarations, which might be a misconfiguration, or might be another bug. There's also a bug splitting macros that exceed the line length, and I'd really like to have separate soft and hard limits.

I'll commit the config (after this lands), but it is definitely still experimental (read: needs to be reviewed by a human and should not just be accepted as truth).

Reply:
Uh... so on further investigation, [our style guide](/Kitware/kwiver/blob/master/doc/code-style.rst) actually says I'm making this up. I'm going to ask around to try to determine which (the guide or my memory) is in error...